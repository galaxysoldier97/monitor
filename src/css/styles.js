import { typography } from '@material-ui/system';

const styles = {
  navigation: {
    paddingBottom: 15,
  },
  title: {
    fontSize: 24,
    fontWeight: typography.fontWeightLight,
    marginBottom: 20,
  },
  paper: {
    padding: 30,
  },
  clear: {
    clear: 'both',
  },
};

export default styles;
