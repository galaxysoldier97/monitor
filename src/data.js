import React from 'react';
import {
  AllInbox,
  AspectRatio,
  Assessment,
  Dialpad,
  LinearScale,
  LocationCity,
  LocationOn,
  Lock,
  Map,
  Settings,
  SignalCellular0Bar,
  SignalCellularAlt,
  SimCard,
  Storefront,
  SupervisorAccount,
  ViewCompact
} from '@material-ui/icons';
import Actions from './actions/Actions';
import {Config} from './config';
import {t} from 'mt-react-library/functions';
import packageConfig from '../package.json';
import {resourcesScopes} from './config/resources/resourcesScopes';
import {Companies} from './config/company';
import {ROUTES} from './config/routes';

export const System = {
  version: packageConfig.version,
  menus: [
    {
      text: t('menu.dashboard'),
      icon: <Assessment />,
      link: '/dashboard',
      requiredPermissions: ['DASHBOARD_READ'],
    },
    {
      text: 'User Management',
      icon: <SupervisorAccount />,
      link: '/access',
      requiredPermissions: ['TEAM_READ', 'USER_READ'],
      category: 'settings',
    },
    {
      text: 'Permission Management',
      icon: <Lock />,
      link: '/permissions',
      requiredPermissions: ['ACL_READ'],
      category: 'settings',
    },
    {
      text: 'Configuration',
      icon: <Settings />,
      link: '/configuration',
      requiredPermissions: ['CFG_READ'],
      category: 'settings',
    },
    {
      id: 'simcards',
      text: t('menu.simcards'),
      icon: <SimCard />,
      link: '/simCard',
      requiredPermissions: [resourcesScopes.simCard.read],
      category: 'equipments',
    },
    {
      id: 'cpes',
      text: t('menu.cpes'),
      icon: <ViewCompact />,
      link: '/cpes',
      requiredPermissions: [resourcesScopes.cpe.read],
      category: 'equipments',
    },
    {
      id: 'ancillaryEquipments',
      text: t('menu.ancillaryEquipments'),
      icon: <AllInbox />,
      link: ROUTES.ancillaryEquipments.path,
      requiredPermissions: [resourcesScopes.ancillaryEquipments.read],
      category: 'equipments',
    },
    {
      id: 'equipmentsAdmin',
      text: t('menu.admin'),
      icon: <Settings />,
      link: '/equipmentsAdmin',
      requiredPermissions: [resourcesScopes.equipmentsAdmin.read],
      category: 'equipments',
    },
    {
      id: 'numbers',
      text: t('menu.numbers'),
      icon: <Dialpad />,
      link: '/numbers',
      requiredPermissions: [resourcesScopes.number.read],
      category: 'resources',
    },
    {
      id: 'rangeNumbers',
      text: t('menu.rangeNumbers'),
      icon: <LinearScale />,
      link: '/rangeNumbers',
      requiredPermissions: [resourcesScopes.number.read],
      excludedPermissions: ['NUMBERS_EQUIPMENTS_DASHBOARD_READ'],
      category: 'resources',
    },
    {
      id: 'ipAddresses',
      text: t('menu.ipAddresses'),
      icon: <LocationOn />,
      link: '/ipAddresses',
      requiredPermissions: [resourcesScopes.number.read],
      excludedPermissions: ['NUMBERS_EQUIPMENTS_DASHBOARD_READ'],
      category: 'resources',
    },
    {
      id: 'resourcesAdmin',
      text: t('menu.admin'),
      icon: <Settings />,
      link: '/resourcesAdmin',
      requiredPermissions: [resourcesScopes.resourceConf.read],
      excludedPermissions: ['NUMBERS_EQUIPMENTS_DASHBOARD_READ'],
      category: 'resources',
    },
    {
      id: 'servicesAccess',
      text: t('menu.services.access'),
      icon: <SignalCellular0Bar />,
      link: '/servicesAccess',
      requiredPermissions: [resourcesScopes.serviceAccess.read],
      category: 'services',
    },
    {
      id: 'servicesComponent',
      text: t('menu.services.component'),
      icon: <SignalCellularAlt />,
      link: '/servicesComponent',
      requiredPermissions: [resourcesScopes.serviceComponent.read],
      category: 'services',
    },
    {
      id: 'servicesAdmin',
      text: t('menu.admin'),
      icon: <Settings />,
      link: Actions.SERVICE_ADMIN.getRoutePath(),
      requiredPermissions: [resourcesScopes.servicesAdmin.read],
      category: 'services',
    },
    {
      id: 'settings',
      text: t('menu.settings'),
      icon: <Settings />,
      link: '/settings',
      requiredPermissions: [resourcesScopes.settings.read],
    },
    {
      id: 'postalAdresses',
      text: t('menu.postalAdresses'),
      icon: <LocationCity />,
      link: '/postaladresses',
      requiredPermissions: [resourcesScopes.address.read],
      category: 'address',
    },
    {
      id: 'building',
      text: t('menu.building'),
      icon: <LocationCity />,
      link: '/buildings',
      requiredPermissions: [resourcesScopes.address.read],
      category: 'address',
    },
    {
      id: 'map',
      text: t('menu.map'),
      icon: <Map />,
      link: '/map',
      requiredPermissions: [resourcesScopes.address.read],
      category: 'address',
      hidden: Config.company !== Companies.mt
    },
    {
      id: 'addressAdmin',
      text: t('menu.admin'),
      icon: <Settings />,
      link: '/addressAdmin',
      requiredPermissions: [resourcesScopes.addressAdmin.read],
      category: 'address',
    },
    {
      id: 'handsets',
      text: t('menu.handsets'),
      icon: <Storefront />,
      link: '/handsets',
      requiredPermissions: [resourcesScopes.handset.read],
      category: 'stock',
    },
    {
      id: 'handsetModels',
      text: t('menu.handsetmodels'),
      icon: <AspectRatio />,
      link: '/handsetmodels',
      requiredPermissions: [resourcesScopes.handset.read],
      category: 'stock',
    },
    {
      id: 'stockAdmin',
      text: t('menu.admin'),
      icon: <Settings />,
      link: '/stockAdmin',
      requiredPermissions: [resourcesScopes.stockAdmin.read],
      category: 'stock',
    },
  ],
};

export const unmUi = {
  access: {
    url: Config.unmBaseUrl + '/accesses',
  },
};

export const Backend = {
  settings: {
    authentication: {
      url: Config.tecrepApiBaseUrl + '/login',
    },
    logout: {
      url: Config.tecrepApiBaseUrl + '/logout',
    },
    users: {
      url: Config.tecrepApiBaseUrl + '/rest/api/user/v1',
    },
    userMenuPreferences: {
      url: Config.tecrepApiBaseUrl + '/rest/api/userMenuPreferences/v1',
      enabled: !!Config.tecrepApiBaseUrl,
    },
    teams: {
      url: Config.tecrepApiBaseUrl + '/rest/api/team/v1',
    },
    permissions: {
      url: Config.tecrepApiBaseUrl + '/rest/api/permissions/v1',
    },
    roles: {
      url: Config.tecrepApiBaseUrl + '/rest/services/userrole',
    },
    configuration: {
      url: Config.tecrepApiBaseUrl + '/rest/api/configuration/v1',
    },
  },
  unm: {
    access: {
      url: Config.unmBaseUrl + '/accesses',
    },
  },
  services: {
    dashboard: {
      url: Config.tecrepApiServicesUrl + '/private/auth/servicesDashboard',
    },
    service: {
      url: Config.tecrepApiServicesUrl + '/private/auth/services',
    },
    access: {
      url: Config.tecrepApiServicesUrl + '/private/auth/accessservices',
    },
    component: {
      url: Config.tecrepApiServicesUrl + '/private/auth/componentservices',
    },
    import: {
      url: Config.tecrepApiServicesUrl + '/private/auth/import',
    },
    activationCode: {
      url: Config.tecrepApiServicesUrl + '/private/auth/activationcodes',
    },
    provisioningTag: {
      url: Config.tecrepApiServicesUrl + '/private/auth/provisioningtags',
    },
    provisioningActions: {
      url: Config.tecrepApiServicesUrl + '/private/auth/provisioningactions',
    },
    tagActivationCodes: {
      url: Config.tecrepApiServicesUrl + '/private/auth/tagactivationcodes',
    },
    product: {
      url: Config.tecrepApiServicesUrl + '/private/auth/products',
    },
    technicalParameters: {
      url: Config.tecrepApiServicesUrl + '/private/auth/technicalparameters',
    },
    activities: {
      url: Config.tecrepApiServicesUrl + '/public/enums/activity',
    },
    componentTypes: {
      url: Config.tecrepApiServicesUrl + '/public/enums/componentType',
    }
  },
  equipments: {
    dashboard: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v1/private/auth/equipmentsDashboard',
    },
    simcards: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/simcards',
    },
    cpes: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/cpes',
    },
    ancillaryEquipments: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/ancillaryequipments',
    },
    warehouses: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/warehouses',
    },
    plmns: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/plmns',
    },
    providers: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/providers',
    },
    inventorypools: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v1/private/auth/inventorypools',
    },
    fileConfigurations: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v1/private/auth/fileConfiguration',
    },
    simcardConfiguration: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/simGenerationConfigurations',
    },
    batches: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v1/private/auth/batch',
    },
    equipmentModel: {
      url: Config.tecrepApiEquipmentsUrl + '/api/v2/private/auth/equipmentModels',
    },
    jobConfiguration: {
      url: `${Config.tecrepApiEquipmentsUrl}/api/v1/private/auth/job/configuration`
    },
    import: {
      url: Config.tecrepApiEquipmentsUrl + '/private/auth/import',
    },
    eSimNotification: {
      url: Config.tecrepApiEquipmentsUrl + '/esim/notification',
    }
  },
  resources: {
    dashboard: {
      url: Config.tecrepApiResourcesUrl + '/private/auth/resourcesDashboard',
    },
    numbers: {
      url: Config.tecrepApiResourcesUrl + '/api/v2/private/auth/numbers',
    },
    ipAddresses: {
      url: Config.tecrepApiResourcesUrl + '/api/v1/private/auth/ipaddresses',
    },
    blockNumbers: {
      url: Config.tecrepApiResourcesUrl + '/api/v2/private/auth/blocknumbers',
    },
    intervalNumbers: {
      url: Config.tecrepApiResourcesUrl + '/api/v2/private/auth/intervalnumbers',
    },
    jobConfiguration: {
      url: Config.tecrepApiResourcesUrl + '/api/v1/private/auth/job/configuration',
    },
    activities: {
      url: Config.tecrepApiResourcesUrl + '/public/enums/activity',
    },
    import: {
      url: Config.tecrepApiResourcesUrl + '/private/auth/import',
    }
  },
  address: {
    buildings: {
      url: Config.tecrepApiAddressUrl + '/private/auth/buildings',
    },
    postalAddress: {
      url: Config.tecrepApiAddressUrl + '/private/auth/postaladdresses',
    },
    buildingStatus: {
      url: Config.tecrepApiAddressUrl + '/private/auth/buildingStatus',
    },
    address: {
      url: Config.tecrepApiAddressUrl + '/private/auth/addresses',
    },
    buildingFlat: {
      url: Config.tecrepApiAddressUrl + '/private/auth/buildingflats',
    },
    externalRefAddresses: {
      url: Config.tecrepApiAddressUrl + '/private/auth/externalrefaddresses',
    },
    accessPoints: {
      url: Config.tecrepApiAddressUrl + '/private/auth/accessPoints',
    },
    ptos: {
      url: Config.tecrepApiAddressUrl + '/private/auth/ptos',
    },
    import: {
      url: Config.tecrepApiAddressUrl + '/private/auth/import',
    },
    streets: {
      url: Config.tecrepApiAddressUrl + '/private/auth/streets',
    },
  },
  stock: {
    handsets: {
      url: Config.tecrepApiStockMgmtUrl + '/private/auth/handsets',
    },
    handsetModels: {
      url: Config.tecrepApiStockMgmtUrl + '/private/auth/handsetmodels',
    },
    warehouses: {
      url: Config.tecrepApiStockMgmtUrl + '/private/auth/warehouses',
    },
    manufacturers: {
      url: Config.tecrepApiStockMgmtUrl + '/private/auth/manufacturers',
    },
  },
  DeliveryFile: {
    import: {
      url: Config.tecrepApiStockMgmtUrl + '/private/auth/import',
    },
  },
};
