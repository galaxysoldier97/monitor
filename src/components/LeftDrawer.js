import React, { useState } from 'react';
import { Collapse, Drawer, List, ListItem, ListItemIcon, ListItemText } from '@material-ui/core';
import { ExpandLess, ExpandMore } from '@material-ui/icons';
import { makeStyles } from '@material-ui/styles';
import { Link } from 'react-router-dom';
import { TecrepUserMenuPreferencesService } from '../services/TecrepUserMenuPreferencesService';
import { Auth } from '../services/Auth';
import { drawerWidth } from '../config';
import { t } from 'mt-react-library/functions';
import { getCompanyLogo, getCompanyName } from '../config/company';
import GreyColor from '@material-ui/core/colors/grey';
import PropTypes from 'prop-types';

const useStyles = makeStyles(theme => ({
  list: {
    width: '100%',
    height: '100%',
    maxWidth: 360,
    display: 'block',
  },
  menuItemIcon: {
    fontSize: 14,
    color: GreyColor[500],
  },
  menuItemText: {
    color: theme.palette.secondary.light + ' !important',
    fontSize: 14,
  },
  avatarDiv: {
    padding: '15px 0 20px 15px',
    height: 60,
  },
  avatarIcon: {
    float: 'left',
    display: 'block',
    marginRight: 15,
    boxShadow: '0px 0px 0px 8px rgba(0,0,0,0.2)',
    padding: 2,
  },
  avatarSpan: {
    paddingTop: 12,
    display: 'block',
    color: theme.palette.secondary.contrastText,
    fontWeight: 300,
    textShadow: '1px 1px #444',
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
    backgroundColor: theme.palette.secondary.dark + ' !important',
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: '0 8px',
    ...theme.mixins.toolbar,
    justifyContent: 'center',
    cursor: 'pointer',
    fontSize: 18,
    backgroundColor: theme.drawer.bgColor || theme.palette.primary.main,
  },
}));

const LeftDrawer = ({drawerOpen, menus, userMenu, refreshUserMenu}) => {
  const classes = useStyles();
  const [userMenuPreferences, setUserMenuPreferences] = useState(userMenu.preferences);

  const login = Auth.getConnectedUser();
  const categories = [
    {id: 'resources', key: 'resources', value: t('menu.resources')},
    {id: 'equipments', key: 'equipments', value: t('menu.equipments')},
    {id: 'services', key: 'services', value: t('menu.services')},
    {id: 'address', key: 'address', value: t('menu.address')},
    {id: 'settings', key: 'settings', value: t('menu.settings')},
    {id: 'stock', key: 'stock', value: t('menu.stock')},
  ];

  const toggleCategory = category => {
    const preferences = {...userMenuPreferences, [category.key]: !userMenuPreferences[category.key]};
    setUserMenuPreferences(preferences);
    if (userMenu.id) {
      return TecrepUserMenuPreferencesService.update({...userMenu, preferences})
      .then(result => result && refreshUserMenu())
      .catch();
    }
    return TecrepUserMenuPreferencesService.create({...userMenu, preferences}, login)
    .then(result => result && refreshUserMenu())
    .catch();
  };

  return (
    <Drawer className={classes.drawer} variant="persistent" open={drawerOpen} classes={{paperAnchorLeft: classes.drawerPaper}}>
      <div className={classes.drawerHeader}>
        <img src={getCompanyLogo()} alt={getCompanyName()} height="49px" />
      </div>
      <List component="nav" className={classes.list}>
        {menus.filter(m => !m.category).map(m => (
          <ListItem button component={Link} to={m.link} key={m.category + m.link}>
            <ListItemIcon id={m.id} className={classes.menuItemIcon}>{m.icon}</ListItemIcon>
            <ListItemText disableTypography className={classes.menuItemText}>{m.text}</ListItemText>
          </ListItem>
        ))}
        {categories.map(category => {
          if (menus.filter(menu => menu.category === category.key).length === 0) {
            return null;
          }
          return (
            <div key={category.key}>
              <ListItem button onClick={() => toggleCategory(category)}>
                <ListItemText id={category.id} disableTypography className={classes.menuItemText}>{category.value}</ListItemText>
                {userMenuPreferences[category.key]
                  ? <ExpandLess className={classes.menuItemText} />
                  : <ExpandMore className={classes.menuItemText} />}
              </ListItem>
              <Collapse in={userMenuPreferences[category.key]} timeout="auto" unmountOnExit>
                <List disablePadding>
                  {menus.filter(m => m.category === category.key).map((menu, index) => (
                    <ListItem key={index} button component={Link} to={menu.link}>
                      <ListItemIcon className={classes.menuItemIcon}>{menu.icon}</ListItemIcon>
                      <ListItemText id={menu.id} disableTypography className={classes.menuItemText}>{menu.text}</ListItemText>
                    </ListItem>
                  ))}
                </List>
              </Collapse>
            </div>);
        })}
      </List>
    </Drawer>
  );
};

LeftDrawer.propTypes = {
  drawerOpen: PropTypes.bool.isRequired,
  menus: PropTypes.array.isRequired,
  userMenu: PropTypes.object.isRequired,
  refreshUserMenu: PropTypes.func.isRequired,
};

export default LeftDrawer;
