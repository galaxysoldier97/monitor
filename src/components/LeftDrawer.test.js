import React from 'react';
import renderer from 'react-test-renderer';
import { createMuiTheme } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/core/styles';
import { IntlProvider } from 'react-intl';
import LeftDrawer from './LeftDrawer';
import { System } from '../data';
import EN from '../i18n/en.json';
import { BrowserRouter } from 'react-router-dom';

test('Left drawer snapshot', () => {
  const theme = createMuiTheme({drawer: {bgColor: '#000'}});
  const component = renderer.create(
    <IntlProvider key="en" locale="en" messages={EN}>
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <LeftDrawer drawerOpen={true} menus={System.menus} refreshUserMenu={() => {
          }} userMenu={{id: null, userId: null, language: 'en', preferences: {}}} /></BrowserRouter>
      </ThemeProvider></IntlProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
