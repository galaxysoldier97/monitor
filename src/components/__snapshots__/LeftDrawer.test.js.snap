// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left drawer snapshot 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-8"
>
  <div
    className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft makeStyles-drawerPaper-9 MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="makeStyles-drawerHeader-10"
    >
      <img
        height="49px"
        src="test-file-stub"
      />
    </div>
    <nav
      className="MuiList-root makeStyles-list-1 MuiList-padding"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        href="/dashboard"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root makeStyles-menuItemIcon-2"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root makeStyles-menuItemText-3"
        >
          Dashboard
        </div>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        href="/settings"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root makeStyles-menuItemIcon-2"
          id="settings"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
              transform="scale(1.2, 1.2)"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root makeStyles-menuItemText-3"
        >
          Settings
        </div>
      </a>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root makeStyles-menuItemText-3"
            id="resources"
          >
            Resources
          </div>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-menuItemText-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root makeStyles-menuItemText-3"
            id="equipments"
          >
            Equipments
          </div>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-menuItemText-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root makeStyles-menuItemText-3"
            id="services"
          >
            Services
          </div>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-menuItemText-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root makeStyles-menuItemText-3"
            id="address"
          >
            Address
          </div>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-menuItemText-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root makeStyles-menuItemText-3"
            id="settings"
          >
            Settings
          </div>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-menuItemText-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root makeStyles-menuItemText-3"
            id="stock"
          >
            Stock
          </div>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-menuItemText-3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
